src/main/java/io/harness/cf/client/api/MetricsProcessor.java

```diff
@@ -39,42 +34,95 @@
   private static final String SDK_LANGUAGE = "SDK_LANGUAGE";
   private static final String SDK_VERSION = "SDK_VERSION";
 
-  private String jarVerion = "";
-
-  private final MetricsApi metricsAPI;
-  private final Cache analyticsCache;
-  private final String environmentID;
-  private final String cluster;
+  private final MetricsCallback callback;
   private final Config config;
+  private final BlockingQueue<MetricEvent> queue;
+  private final MetricsApi api;
+
+  private String jarVersion = "";
+
+  @Setter private String environmentID;
+  @Setter private String cluster;
```

Comment:
Is it possible to get rid of setters and make the object immutable?

Reply:
for the moment no, because there are some upcoming features